libsolidity/formal/SMTChecker.cpp

```diff
@@ -701,6 +704,19 @@ void SMTChecker::resetVariables(vector<Declaration const*> _variables)
 	}
 }
 
+void SMTChecker::mergeVariables(vector<Declaration const*> _variables, smt::Expression _condition, VariableSequenceCounters& _countersEndTrue, VariableSequenceCounters& _countersEndFalse)
+{
+	for (auto const* decl: _variables)
+	{
+		newValue(*decl);
+		int trueCounter = _countersEndTrue.at(decl);
+		int falseCounter = _countersEndFalse.at(decl);
+		smt::Expression e = currentValue(*decl) == smt::Expression::ite(_condition, valueAtSequence(*decl, trueCounter), valueAtSequence(*decl, falseCounter));
+		m_interface->addAssertion(e);
```

Comment:
I think there should be a helper function for `addAssertion(currentValue(*decl) == ...)`

Reply:
There's a function `defineExpr`, but the first argument is an `Expression`, and not `smt::Expression`. Overloaded the function to work with `smt::Expression` as well.

Reply:
Ah ok, looking at `::assignment(...)` - which is basically the same thing. There, the following is done:
`m_interface->addAssertion(newValue(_variable) == _value);`

Reply:
Yes, good, that's how I have it currently.